comms/dht/src/discovery/service.rs

```diff
@@ -420,25 +408,9 @@ impl DhtDiscoveryService {
     async fn send_discover(
         &mut self,
         nonce: u64,
-        discovery_request: DiscoverPeerRequest,
+        dest_public_key: Box<CommsPublicKey>,
     ) -> Result<(), DhtDiscoveryError>
     {
-        let DiscoverPeerRequest {
```

Comment:
I'm not sure I agree with the removal of the DiscoverPeerRequest, except if it will be brought back at a later stage when discovery becomes more stable. These changes will disable the more efficient discovery using the node_id and will only allow discovery using flooding on the network. This can become expensive when the network is large.

Reply:
I agree we need to find a good way to propagate encrypted single-destination messages across the network without propagating to every node while still preserving privacy. There was no way to do this before so I removed the extra cruft in the discover_peer call for now. I do plan to readd it, however when it gets added back in, it needs to make sure that "closer" nodes are online and possibly only choose out of it's currently connected neighbours to propagate the message. TLDR it's quite tricky to ensure that private targeted discovery is reliable on our current testnet network without flooding so I wanted to take a step back and add a working solution later - as you suggested (elsewhere) perhaps using the hamming distance for these kind of messages would allow privacy to be preserved while still moving the message in the right direction.